### ISSUE TITLE
[BUG] std.join doesn't work the same as  c++ version's

### ISSUE THREAD

**Issue by PureWhiteWu** [google/go-jsonnet#171]
Hi, it's me again, I found another thing which behaviours not the same as the c++'s version.

![image](https://user-images.githubusercontent.com/3314335/34824526-ce8665ce-f709-11e7-8342-2f1d84e05fe2.png)

I think this is because the  go join function doesn't behaviours the same as the c++ version's.
In  the c++'s version, if the expression result is null, it will be  ignored.
@sbarzowski 
@sparkprime

**Comment by PureWhiteWu** [comment]
The command is: 
```
./jsonnet -e "std.join(\".\", [\"1\", \"2\", if 0 > 0 then 0])"
```

**Comment by PureWhiteWu** [comment]
I found the bug, but i don't know how to fix it so far.
![image](https://user-images.githubusercontent.com/3314335/34825172-574804ce-f70c-11e7-83c5-2bd8602cafe7.png)

![image](https://user-images.githubusercontent.com/3314335/34825198-7bae4fb2-f70c-11e7-93eb-1db8196af96e.png)

**Comment by sparkprime** [comment]
I noticed this just before you did and fixed it in #170 :)

**Comment by PureWhiteWu** [comment]
wow great!
